librz/core/cdebug.c

```diff
@@ -639,3 +639,273 @@ RZ_API RzCmdStatus rz_core_debug_plugins_print(RzCore *core, RzCmdStateOutput *s
+	mul = findMinMax(maps, &min, &max, 0, width);
+	ut64 last = min;
+	if (min != -1 && mul != 0) {
+		const char *color_prefix = ""; // Color escape code prefixed to string (address coloring)
+		const char *color_suffix = ""; // Color escape code appended to end of string
+		const char *fmtstr;
+		char humansz[8]; // Holds the human formatted size string [124K]
+		int skip = 0; // Number of maps to skip when re-calculating the minmax
+		rz_list_foreach (maps, iter, map) {
+			rz_num_units(humansz, sizeof(humansz), map->size); // Convert map size to human readable string
+			if (colors) {
+				color_suffix = Color_RESET;
+				if ((map->perm & 2) && (map->perm & 1)) { // Writable & Executable
+					color_prefix = pal->widget_sel;
+				} else if (map->perm & 2) { // Writable
+					color_prefix = pal->graph_false;
+				} else if (map->perm & 1) { // Executable
+					color_prefix = pal->graph_true;
+				} else {
+					color_prefix = "";
+					color_suffix = "";
+				}
+			} else {
+				color_prefix = "";
+				color_suffix = "";
+			}
+			if ((map->addr - last) > UT32_MAX) { // TODO: Comment what this is for
+				mul = findMinMax(maps, &min, &max, skip, width); //  Recalculate minmax
+			}
+			skip++;
+			fmtstr = dbg->bits & RZ_SYS_BITS_64 // Prefix formatting string (before bar)
+				? "map %4.8s %c %s0x%016" PFMT64x "%s |"
+				: "map %4.8s %c %s0x%08" PFMT64x "%s |";
+			dbg->cb_printf(fmtstr, humansz,
+				(addr >= map->addr &&
+					addr < map->addr_end)
+					? '*'
+					: '-',
+				color_prefix, map->addr, color_suffix); // * indicates map is within our current sought offset
+			int col;
+			for (col = 0; col < width; col++) { // Iterate over the available width/columns for bar graph
+				ut64 pos = min + (col * mul); // Current address space to check
+				ut64 npos = min + ((col + 1) * mul); // Next address space to check
+				if (map->addr < npos && map->addr_end > pos) {
+					dbg->cb_printf("#"); // TODO: Comment what a # represents
+				} else {
+					dbg->cb_printf("-");
+				}
+			}
+			fmtstr = dbg->bits & RZ_SYS_BITS_64 ? // Suffix formatting string (after bar)
+				"| %s0x%016" PFMT64x "%s %s %s\n"
+							    : "| %s0x%08" PFMT64x "%s %s %s\n";
+			dbg->cb_printf(fmtstr, color_prefix, map->addr_end, color_suffix,
+				rz_str_rwx_i(map->perm), map->name);
+			last = map->addr;
+		}
+	}
+}
+
+RZ_API void rz_debug_map_list_visual(RzDebug *dbg, ut64 addr, const char *input, int colors) {
+	if (!dbg) {
+		return;
+	}
+	int i;
+	for (i = 0; i < 2; i++) { // Iterate over dbg::maps and dbg::maps_user
+		//		RzList *maps = (i == 0) ? dbg->maps : dbg->maps_user;
+		RzList *maps = rz_debug_map_list(dbg, (bool)i);
+		if (!maps) {
+			continue;
+		}
+		RzListIter *iter;
+		RzDebugMap *map;
+		if (input[1] == '.') { // "dm=." Only show map overlapping current offset
+			dbg->cb_printf("TODO:\n");
+			rz_list_foreach (maps, iter, map) {
+				if (addr >= map->addr && addr < map->addr_end) {
+					// print_debug_map_ascii_art (dbg, map);
```

Comment:
Why this part is commented?

Reply:
This is code for `dm=.` command and it was not fully implemented (therefore the `TODO` printf) even before the refactor. I can just remove this part since this code is not reachable after shifting this command to the new shell.

Reply:
Yes, I think it's better to just remove that. If you think this is a feature is worth adding - just open an issue. But better to remove this dead code and TODO printing.